PeriodogramAnalysis/PowerSpectrumAnalysis.py

```diff
@@ -19,246 +24,254 @@ class SpectrumAnalysis(OperatorMixin):
 
         Parameters:
         -----------
-        signal : SignalType
+        signal : Generator
             Input signal to be analyzed.
 
         Returns:
         --------
-        tuple
-            PSD dictionary, excluded channels, number of chunks, band display range, dB conversion flag, spectrogram dictionary, frequency limit.
+        psd_dict:
+            PSD dictionary
+        spec_dict
+            spectrum dictionary
         """
-        for idx, signal_piece in enumerate(signal):
-            if idx >= 1:
-                break
-            self.num_channels = signal_piece.number_of_channels
-            num_of_chunks = idx + 1
-
+        psd_dict: DictType = defaultdict(dict)
+        spec_dict: DictType = defaultdict(dict)
+        for chunk_index, signal_piece in enumerate(signal):
             # Plot spectrum using different methods
-            psd_dict = self.computing_multi_spectrum(signal_piece, self.exclude_channels, num_of_chunks, self.win_sec, self.convert_db)
+            psd_dict[chunk_index] = self.computing_multi_spectrum(signal_piece)
             # Plot spectrogram
-            spec_dict = self.computing_spectrum(signal_piece, self.exclude_channels, num_of_chunks, self.nperseg, self.noverlap)
-
-        return psd_dict, self.exclude_channels, num_of_chunks, self.band_display, self.convert_db, spec_dict, self.frequency_limit
-
-    def computing_multi_spectrum(self, signal, exclude_channel_list, num_of_chunks, win_sec, dB):
-        win = win_sec * signal.rate
-        psd_dict = {}
-
-        for chunk in range(num_of_chunks):
-            if chunk not in psd_dict:
-                psd_dict[chunk] = {}
-            for channel in range(self.num_channels):
-                if channel not in exclude_channel_list:
-                    freqs_per, psd_per = sig.periodogram(signal.data[:, channel], signal.rate)
-                    freqs_welch, psd_welch = sig.welch(signal.data[:, channel], fs=signal.rate, nperseg=win)
-                    psd_mt, freqs_mt = multitaper_psd(signal.data[:, channel], signal.rate)
-                    sharey = False
-
-                    if dB:
-                        psd_per = 10 * np.log10(psd_per)
-                        psd_welch = 10 * np.log10(psd_welch)
-                        psd_mt = 10 * np.log10(psd_mt)
-                        sharey = True
-
-                    psd_dict[chunk][channel] = {
-                        'freqs_per': freqs_per,
-                        'freqs_welch': freqs_welch,
-                        'freqs_mt': freqs_mt,
-                        'psd_per': psd_per,
-                        'psd_welch': psd_welch,
-                        'psd_mt': psd_mt,
-                        'sharey': sharey
-                    }
+            spec_dict[chunk_index] = self.computing_spectrum(signal_piece)
+
+        return psd_dict, spec_dict
+
+    def computing_multi_spectrum(self, signal: Signal) -> Dict[int, Dict[str, Any]]:
```

Comment:
????

Reply:
This repeats all the spectrum analysis from other operator

Reply:
Have got it removed, I moved all PSD computing methods to a separate file, in commit [bc7e61d]